### DISCUSSION TITLE
[Node Operator Question] Since the dencun upgrade went live on goerli, my op-goerli node has halted

### DISCUSSION THREAD

**Discussion by Shankusu993** [ethereum-optimism/developers#200] [Node Operators]
### Are you running the most up to date node software?

- [X] Yes

### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Since the dencun upgrade went live, my optimism goerli testnet node has been halted. I have already verified that the L1 node my op-node connects to is upgraded (it's from alchemy btw), but still the `op-geth` is stuck and `op-node` gives following logs:
![image](https://github.com/ethereum-optimism/developers/assets/43635589/1c1dea28-9f36-4b5d-b01a-06feea429798)

### Protocol Description

- op-geth version = v1.101304.2
- op-node version = v1.4.0
- network name = op-goerli
- chainid = 420
- L1 chain = goerli (chain id 5)

### Node Logs

- please refer the attached pic of op-node logs in the description

**Comment by imtipi** [comment]
seems Optimism Goerli has been completely deprecated,etherscan shows it stopped produce blocks one day ago
![image](https://github.com/ethereum-optimism/developers/assets/27230181/ca6e4590-1613-4b7f-8fbe-7b0ee90e1c53)
check https://github.com/ethereum-optimism/developers/discussions/107 for more info
cc @sbvegan can you confirm Optimism Goerli is completely stopped or not?

**Reply by Shankusu993** [reply]
@imtipi Thanks for the response, I'm well aware of #107  and have a plan in-place to shift to sepolia by 2nd week of feb. As for etherscan, I'd guess that etherscan is facing the same issue as me, their node, the source to their data, must have halted aswell. Optimism Goerli network isn't completely stopped yet, some third party L2 nodes shows new blocks being added and also a simple query to optimism's sequencer node ([https://goerli-sequencer.optimism.io/](https://goerli-sequencer.optimism.io/)) shows the chain is indeed still moving.

**Comment by Chomtana** [comment]
I'm not sure if we need to modify rollup.json to include some more field about decun time

You can use this script to check if your chain is on the right fork or not

```bash
#!/bin/bash

# RPCs
OP_RETH="<op-geth-rpc-endpoint>"
BASE="https://mainnet.base.org"

NOW=$(cast block-number --rpc-url $OP_RETH)
BLOCK_DIFF=$(diff <(cast block $NOW --rpc-url $OP_RETH) <(cast block $NOW --rpc-url $BASE))

# Check if BLOCK_DIFF is empty
if [[ -z "$BLOCK_DIFF" ]]; then
        echo "No diff in block #$NOW between op-reth and canonical base mainnet RPC."
else
        echo "-----------------------------------------------------------------------"
        echo " -> ERROR! Diff in block #$NOW between op-reth and canonical base mainnet RPC"
        echo "$BLOCK_DIFF"
        echo "-----------------------------------------------------------------------"
fi
```

Source: https://github.com/paradigmxyz/reth/pull/6050#issuecomment-1892897948

**Reply by Shankusu993** [reply]
Hi @Chomtana , generalised the script a bit and ran it for my op-goerli node, I got  No diff output, so my chain seem to not have forked.

- Sequencer URL I used : "https://goerli.optimism.io"
- output I got 
![image](https://github.com/ethereum-optimism/developers/assets/43635589/11b21d7c-e2b3-4fe8-8ce8-6649ae848d59)
- The slightly more generalized script :




```
#!/bin/bash

# RPCs
OP_GETH="<op-geth-rpc-endpoint>"
SEQUENCER="<sequencer-rpc-endpoint>"

NOW=$(cast block-number --rpc-url $OP_GETH)
BLOCK_DIFF=$(diff <(cast block $NOW --rpc-url $OP_GETH) <(cast block $NOW --rpc-url $SEQUENCER))

# Check if BLOCK_DIFF is empty
if [[ -z "$BLOCK_DIFF" ]]; then
        echo "No diff in block #$NOW between op-geth and canonical RPC."
else
        echo "-----------------------------------------------------------------------"
        echo " -> ERROR! Diff in block #$NOW between op-geth and canonical RPC"
        echo "$BLOCK_DIFF"
        echo "-----------------------------------------------------------------------"
fi
```

- My L2 node is still dead (`op-geth` doesn't progress at all but doesn't have error either), the following is the only error/warning log I see on `op-node`:
![image](https://github.com/ethereum-optimism/developers/assets/43635589/7d890836-dca3-47c4-a0cb-e79fc099812c)

**Reply by Shankusu993** [reply]
just for an update, I also tried a couple of different things:
- Restart the geth and node (because why not :p)
- Rewind chain (op-geth) to last finalized l2 block and restart both the components
- Rewind op-geth to a further earlier block in-process invalidating safe and finalized heads and then restarting both the components
- Restart from an older checkpoint which I'm sure of was correct

None of them seem to do the trick currently
